src/Storages/RabbitMQ/StorageRabbitMQ.cpp

```diff
@@ -242,6 +242,8 @@ ContextMutablePtr StorageRabbitMQ::addSettings(ContextPtr local_context) const
     modified_context->setSetting("input_format_skip_unknown_fields", true);
     modified_context->setSetting("input_format_allow_errors_ratio", 0.);
     modified_context->setSetting("input_format_allow_errors_num", rabbitmq_settings->rabbitmq_skip_broken_messages.value);
+    /// Since we are reusing the same context for all queries executed simultaneously, we don't want to used shared `analyze_count`
+    modified_context->setSetting("max_analyze_depth", 0);
```

Comment:
Seems like the same issue is possible with StorageNATS, but maybe we don't have enough tests to reproduce it:
https://github.com/ClickHouse/ClickHouse/blob/7a4be3b748c7e9d0ca3ad06c1c172e977b6537af/src/Storages/NATS/StorageNATS.cpp#L91
cc: @kssenii